offline/cable_LUC_EXPT.F90

```diff
@@ -935,20 +969,20 @@ SUBROUTINE READ_LUH2(LUC_EXPT)
     endif
 
     ! Adjust transition areas based on primary wooded fraction
+    ! Note that only the four transitions ptos, ptog, gtos, and stog are 
```

Comment:
Thanks for including a comment

Reply:
Actually - on further thought - do `ptoc`, `ptoq`, `stoc`, `stoq`, `ctos` and `qtos` need a wood fraction or not?  These transitions are not within the grass type.

They may actually need to be `ptoc = ptoc * ptog` etc. here. - but this connects into exactly how the updated LUH2 processing works.

(for the moment it's probably okay - the code will work in serial mode which is what you've used for TRENDY v13)

Reply:
It should be fine like this. They are included in the four main classes. For example, ptoc and ptoq are already accounted for through ptog. stoq is in included in stog. ctos is included in gtos. etc. 

After TRENDY I will also commit the LUH2 postprocesssing script that clarifies this in more detail.

Reply:
unfortunately no.

For example (albeit a really odd situation) 
- let grass=crop=pasture=0 at the start of the time step
- let there be (only) a ptog which is a combination of ptoc and ptoq

if ptoq is used to evaluate the %pasture fraction (line 1471 in POPLUC.F90)  without the wood_fraction include, then you'd get pasture = grass (since you hit the hard limit) and crop = 0  - that's not the intent.

you can create other quirks (e.g. combinations of stog and stoc can lead to crop=0 error).

Reply:
Not sure I follow. Let's say grass = 0 at the beginning. And you have a conversion of ptog = 0.05, ptoq = 0.025, and ptoc = 0.025 and no other transitions. Shouldn't that work?

The way the LUH2 processing script works is that ptog is the sum of all transitions from primary to all grass types (so ptog = ptoq + ptoc in this case). So it should be constrained in that way. But there might edge cases!

Reply:
But say wood_frac is 0.4 - then ptog becomes 0.02 but ptoq and ptoc stay at 0.025.

The sequence is then grass->0.02, pasture -> 0.02 (because pasture<=grass at line 1471) and crop -> 0.0 (because crop < grass-pasture at line 1473)

This is an edge case - and one that is highly unlikely since if wood_frac<1 then I can't see how you get grass=0 - but you could conceive cases with a small amount of primn_grass and a single year where an unmodified ptoc and/or ptoq would be enough to hit a limit (e.g. rapid deforestation in the Amazon where CABLE-POP thinks there's little/no grass but LUH2 did)

IMPORTANTLY - all this is for the future (so not a blocker to this PR)

Reply:
After further thought - this is less of an issue (possibly even a complete non-issue) that I first thought.  The trick here is to remember that (since LUH2 is internal consistent) you can't have the case above, you have to have some LUH2 state available to transition away from.

If we have ptog = A >0, ptoq = A1 >0, ptoc = A2 >0 (A=A1+A2) with no other transitions and 0<WF<1 then that implies that
- LUH2 primary at start is at least A
- POPLUC PF at start  PF0>=WF A and POPLUC PG at start PG0>= A (1-WF)
- time stepping then givesPF -> PF0-WF A>=0, PG -> PG0-(1-WF)*A >=0
- GR = PG0 + CR0 + PA0 -> PG0 + CR0 + PA0 - (1-WF)*A + A1 + A2 = GR0 + WF*A >0
- with CR->CR0+A1 (always < GR, since PG0-(1-WF)*A>0, PA0>0 & A2>0)
- and PA-> PA0+A2 (always < GR-CR)

I haven't followed all the other cases through but I think it's okay

***EXCEPT*** for the awkward case of transitions from primary to rangeland and then to other types (see next)

Reply:
OK. Good that it seems to be Ok for most cases. Primaryn to rangeland and then to let's say cropland should be OK for most cases. Because now we account for rtoc conversions that is not multiplied by woodfrac. For example in North Australia we currently have 40% trees, 60% grass. If we have 5% rtoc this will turn 5% of that grass into crops. If we have 65% rtoc we turn 60% grass into crops, but not more. That would be one of the edge cases where our unknown (and in that case incorrect) woodfrac does not allow us to get the right LU areas.